### DISCUSSION TITLE
In DataTable allow unsorting does not work as expected

### DISCUSSION THREAD

**Discussion by hgfrey** [primefaces/community#1641] [PrimeFaces] [answered]
I would have expected that it would be possible to restore the original order with allowUnsorting. That's not the case. The table shows the last sorted state. How can I restore the original state to a table that has been changed to unsorted?

Thank you

**Comment by sja-cslab** [comment]
You're looking for [removableSort ](https://primereact.org/datatable/#api.DataTable.props.removableSort)i guess.

**Reply by hgfrey** [reply]
I think in Primefaces allowUnsorting is equivalent to removableSort in PrimeReact. And allowUnsorting is turned on. But if I take away the sorting, the last sorting remains.

**Reply by sja-cslab** [reply]
My bad - read wrong Category.
I just noticed that the [Sort Demo on Primefaces](https://www.primefaces.org/showcase/ui/data/datatable/sort.xhtml?jfwid=5cb5a) has the same behaviour.
However the [Sort on PrimReact](https://primereact.org/datatable/#removable_sort) works differently.
I'm not sure which one is correct.

@melloware You always seem to have a general idea about the different libs - can you tell us something about that?

**Reply by melloware** [reply]
I personally prefer the sort behavior of PrimeFaces.   The other 3 libs handle it a little differently.

**Reply by sja-cslab** [reply]
Isn't prime trying to keep them sync in behavior?

**Reply by melloware** [reply]
Yes they are but PrimeFaces JSF has had this behavior first and I guess PrimeTek decided not to model the JS libs the same as we did in PF?

**Reply by melloware** [reply]
@sja-cslab which behavior do you like?  JSF or the JS libs way?

**Reply by sja-cslab** [reply, marked as answer]
@melloware I prefer the way it is in PrimeReact.
If I pass values that may be pre-sorted, I expect that ordering if the Datatable sort is removed.
